django/db/backends/oracle/schema.py

```diff
@@ -56,22 +63,35 @@ def alter_field(self, model, old_field, new_field, strict=False):
             # SQLite-ish workaround
             if 'ORA-22858' in description or 'ORA-22859' in description:
                 self._alter_field_type_workaround(model, old_field, new_field)
+            # If we're changing identity column type to non-numeric type
+            # we need to drop identity previous
+            elif 'ORA-30675' in description:
+                self._drop_identity(model._meta.db_table, old_field.column)
+                self.alter_field(model, old_field, new_field, strict)
+            # Modifying identity column isn't supported in Oracle. If we're
```

Comment:
If a primary key column is changing to an identity column, drop the primary key first.